sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -0,0 +1,929 @@
+  [[sycl::reqd_sub_group_size(SG_SIZE)]] {
+   const auto global_idx = item.get_global_id(0);
+   const auto global_idy = item.get_global_id(1);
+   const auto sg_startx = global_idx - item.get_local_id(0);
+   const auto sg_starty = global_idy - item.get_local_id(1);
+   sub_group sg = item.get_sub_group();
+   joint_matrix<sub_group, int8_t, use::a, tM, tK, layout::row_major> tA;
+   joint_matrix<sub_group, int8_t, use::b, tK, tN, layout::row_major> tB;
+   joint_matrix<sub_group, int32_t, use::accumulator, tM, tN> tC;
+   joint_matrix_fill(sg, tC, 0);
+   for (int k = 0; k < K; k += tK) {
+     joint_matrix_load(sg, tA,
+          multi_ptr<int8_t, sycl::access::address_space::global_space>(memA) +
+          sg_startx * tM * K + k, K);
+     joint_matrix_load(sg, tB,
+          multi_ptr<int8_t, sycl::access::address_space::global_space>(memB) +
+          k * N + sg_starty/SG_SIZE*tN, N);
+     tC = joint_matrix_mad(sg, tA, tB, tC);
+   }
+   joint_matrix_apply(sg, tC, [=](int8_t x) {
+    x *= alpha;
+   });
+   joint_matrix_store(sg, tC,
+        multi_ptr<int32_t, sycl::access::address_space::global_space>(memC) +
+        sg_startx * tM * N + sg_starty/SG_SIZE*tN, N, layout::row_major);
+}).wait();
+```
+
+=== Query Interface
+Most devices support only certain values for the `Rows` and `Cols`
+template parameters and only certain types for the `T` template
+parameter. Moreover, most devices support only certain combinations of
+these template parameter for the A, B, C, and D matrices in the
+`joint_matrix_mad` function (see Appendix: Supported Combinations Per
+Hardware). This extension adds two query APIs that can be used to
+determine the set of legal parameters for a particular device. One
+form provides `constexpr` values for these parameters, which can be
+used when the application knows the specific device architecture on
+which it will run. The other form uses the standard information
+descriptor queries for the device object.
+
+The description below uses the terms `M`, `N`, and `K` to identify the
+matrix dimensions of a multiply and add operation `D = C + A*B`. The
+`D` and `C` matrices are `M` rows by `N` columns. The `A` matrix is
+`M` rows by `K` columns, and the `B` matrix is `K` rows by `N` columns.
+
+==== Compile-Time Query
+This returns `constexpr` values to use in `joint_matrix` template
+arguments but depends on an enumeration of the matrix hardware (See
+`sycl::ext::oneapi::experimental::architecture`) in the
+link:../sycl_ext_oneapi_device_architecture.asciidoc[sycl_ext_oneapi_device_architecture]
+extension that can be tested.
+The compile-time query interface proposed here consists of two
+functionalities:
+
+- Validation: at compile time, the validation functionality informs
+  the user whether a specific combination is valid or not. This takes
+  place when the user specifies all template parameters.
+
+- Default values: this provides a default shape if the user does not
```

Comment:
Can you make the implementation choose default values with portability in mind? e.g return a value for XMX that matches the default value for AMX (If I remember correctly there is a unique case satisfying this?).

Reply:
There is currently no such case that satisfies XMX of DG2 and XMX of PVC. But it can be added. Currently the default is the max.